src/libraries/Microsoft.PowerFx.Core/Functions/FunctionArgValidators/ConnectedDataSourceInfoArgValidator.cs

```diff
@@ -56,6 +56,11 @@ private bool TryGetDsInfo(CallNode callNode, TexlBinding binding, out IExternalD
                 return false;
             }
 
+            if (function is UserDefinedFunction udf)
+            {
+                return udf.TryGetExternalDataSource(out dsInfo);
+            }
+
             var success = function.TryGetDataSource(callNode, binding, out var external);
```

Comment:
>TryGetDataSource [](http://example.com/codeflow?start=35&length=16)

is there a reason why we cant override this method in UDF to return correct result ?